File: beecrowd/2669.py
import string

alfa = list(string.ascii_lowercase)
dic = {}
for i in range(26):
    dic[alfa[i]] = i+1

def calc_peso(sub_str):
    if len(sub_str) != 0:
        peso = 0
        for s in sub_str:
            peso += dic[s]
        return peso
    return 0

#==================================#

s = list(input())
pesos = []
tam = len(s)

for i in range(tam+1):
    for j in range(tam):
        sub = s[j:j+i]
        peso = calc_peso(sub)
        if peso not in pesos:
            pesos.append(peso)

print(len(pesos)-1)

File: beecrowd/2669.cpp
#include <bits/stdc++.h>

using namespace std;

int calc_peso(string sub_str){
    int peso = 0;
    for (int i = 0; i < sub_str.size(); i++){
        peso += int(sub_str[i])-96;
    }
    return peso;
}

int main(){

    string s, sub;
    cin >> s;
    vector<int> pesos;
    int i, j, peso, tam = s.size();

    for (i = 1; i <= tam; i++){
        for (j = 0; j < tam; j++){
            if ((j+i) <= tam){
                sub = s.substr(j,i);
                peso = calc_peso(sub);
                if ( std::find(pesos.begin(), pesos.end(), peso) == pesos.end() )
                    pesos.push_back(peso);
            }
        }
    }

    cout << pesos.size() << endl;

    return 0;
}
